### ISSUE TITLE
[SUGGESTION] Can we also include @types/xyz dependencies in align-deps

### ISSUE THREAD

**Issue by SparshaSaha** [microsoft/rnx-kit#2398]
### What happened?

I have seen quite often that people who use rnx-kit/align-deps, update their dependencies properly as align-deps does it for them. But they often forget to update the type dependencies.
For example,
with React: 18.x.x we should use @types/React: 18.x.x.
with ReactNative: 0.71.x, we do not need types/react-native as types are already included.
There could be merit in configuring align-deps to factor in these dependencies as well.

### Affected Package

@rnx-kit/align-deps

### Version

2.2.0

### Which platforms are you seeing this issue on?

- [ ] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### System Information

```true
NA
```

### Steps to Reproduce

NA

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kelset** [comment]
submit a PR?

**Comment by SparshaSaha** [comment]
@kelset sure I can submit a PR but before I did that i wanted to confirm if you folks also think that this will be a meaningful addition.

**Comment by tido64** [comment]
I think it makes sense to add it. My only concern is that we should _not_ make the `react` capability take a dependency on it. It's not a given that people use TypeScript.

**Comment by SparshaSaha** [comment]
Yeah. Makes sense. We can maybe add "types" as a capability. If "types" is listed as a capability, the tool will check/include typings for required packages. What do you think?

**Comment by tido64** [comment]
> Yeah. Makes sense. We can maybe add "types" as a capability. If "types" is listed as a capability, the tool will check/include typings for required packages. What do you think?

I think that will require a lot more work and no longer makes the core engine generic. If we just add a `types/react` capability, the types package will be checked when you run in "vigilant" mode. I find that most people do this anyway to catch packages that they forgot to add.

**Comment by tido64** [comment]
Closed as duplicate of https://github.com/microsoft/rnx-kit/issues/269
